cpp/src/spatial/haversine.cu

```diff
@@ -0,0 +1,153 @@
+// /*
+//  * Copyright (c) 2020, NVIDIA CORPORATION.
+//  *
+//  * Licensed under the Apache License, Version 2.0 (the "License");
+//  * you may not use this file except in compliance with the License.
+//  * You may obtain a copy of the License at
+//  *
+//  *     http://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  * Unless required by applicable law or agreed to in writing, software
+//  * distributed under the License is distributed on an "AS IS" BASIS,
+//  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  * See the License for the specific language governing permissions and
+//  * limitations under the License.
+//  */
+
+#include <cmath>
+#include <memory>
+#include <type_traits>
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cuspatial/error.hpp>
+#include "rmm/mr/device/device_memory_resource.hpp"
+#include "rmm/thrust_rmm_allocator.h"
+#include <cuspatial/constants.hpp>
```

Comment:
Put a blank line between different groups of headers, because when we enable clang-format it will sort within groups alphabetically. Also, avoid quotes.

```suggestion
#include <cmath>
#include <memory>
#include <type_traits>

#include <rmm/mr/device/device_memory_resource.hpp>
#include <rmm/thrust_rmm_allocator.h>

#include <cudf/column/column.hpp>
#include <cudf/column/column_factories.hpp>
#include <cudf/column/column_view.hpp>
#include <cudf/utilities/type_dispatcher.hpp>

#include <cuspatial/constants.hpp>
#include <cuspatial/error.hpp>
```